utility/gov.go

```diff
@@ -10,1798 +12,475 @@ import (
-	store := u.Store()
-	fee, er := store.GetMessageUnpauseValidatorFee()
-	if er != nil {
-		return nil, types.ErrGetParam(types.MessageUnpauseValidatorFee, er)
-	}
-	return types.StringToBigInt(fee)
+	return u.getBigIntParam(types.MessageUnpauseValidatorFee)
 }
 
 func (u *UtilityContext) GetMessageStakeServiceNodeFee() (*big.Int, types.Error) {
-	store := u.Store()
-	fee, er := store.GetMessageStakeServiceNodeFee()
-	if er != nil {
-		return nil, types.ErrGetParam(types.MessageStakeServiceNodeFee, er)
-	}
-	return types.StringToBigInt(fee)
+	return u.getBigIntParam(types.MessageStakeServiceNodeFee)
 }
 
 func (u *UtilityContext) GetMessageEditStakeServiceNodeFee() (*big.Int, types.Error) {
-	store := u.Store()
-	fee, er := store.GetMessageEditStakeServiceNodeFee()
-	if er != nil {
-		return nil, types.ErrGetParam(types.MessageEditStakeServiceNodeFee, er)
-	}
-	return types.StringToBigInt(fee)
+	return u.getBigIntParam(types.MessageEditStakeServiceNodeFee)
 }
 
 func (u *UtilityContext) GetMessageUnstakeServiceNodeFee() (*big.Int, types.Error) {
-	store := u.Store()
-	fee, er := store.GetMessageUnstakeServiceNodeFee()
-	if er != nil {
-		return nil, types.ErrGetParam(types.MessageUnstakeServiceNodeFee, er)
-	}
-	return types.StringToBigInt(fee)
+	return u.getBigIntParam(types.MessageUnstakeServiceNodeFee)
 }
 
 func (u *UtilityContext) GetMessagePauseServiceNodeFee() (*big.Int, types.Error) {
-	store := u.Store()
-	fee, er := store.GetMessagePauseServiceNodeFee()
-	if er != nil {
-		return nil, types.ErrGetParam(types.MessagePauseServiceNodeFee, er)
-	}
-	return types.StringToBigInt(fee)
+	return u.getBigIntParam(types.MessagePauseServiceNodeFee)
 }
 
 func (u *UtilityContext) GetMessageUnpauseServiceNodeFee() (*big.Int, types.Error) {
-	store := u.Store()
-	fee, er := store.GetMessageUnpauseServiceNodeFee()
-	if er != nil {
-		return nil, types.ErrGetParam(types.MessageUnpauseServiceNodeFee, er)
-	}
-	return types.StringToBigInt(fee)
+	return u.getBigIntParam(types.MessageUnpauseServiceNodeFee)
 }
 
 func (u *UtilityContext) GetMessageChangeParameterFee() (*big.Int, types.Error) {
-	store := u.Store()
-	fee, er := store.GetMessageChangeParameterFee()
-	if er != nil {
-		return nil, types.ErrGetParam(types.MessageChangeParameterFee, er)
-	}
-	return types.StringToBigInt(fee)
+	return u.getBigIntParam(types.MessageChangeParameterFee)
 }
 
 func (u *UtilityContext) GetDoubleSignFeeOwner() (owner []byte, err types.Error) {
-	store := u.Store()
-	owner, er := store.GetMessageDoubleSignFeeOwner()
-	if er != nil {
-		return nil, types.ErrGetParam(types.DoubleSignBurnPercentageParamName, er)
-	}
-	return owner, nil
+	return u.getByteArrayParam(types.DoubleSignBurnPercentageParamName)
 }
 
 func (u *UtilityContext) GetParamOwner(paramName string) ([]byte, error) {
+	//DISCUSS (@deblasis): here we could potentially leverage the struct tags in gov.proto by specifying an `owner` key
```

Comment:
This could perhaps be optimized further by adding the appropriate attributes in our new `pokt` struct tag in `gov.proto`.